File: lv1/문자열 내 마음대로 정렬하기.py
def solution(strings, n):
    answer = []
    dir = {}
    for index, el in enumerate(strings):
      if ord(el[n]) not in dir:
        dir[ord(el[n])] = [el]
      else:
        dir[ord(el[n])].append(el)
    dirArr = sorted(dir.items())
    for el in dirArr:
      el[1].sort()
      answer.append(el[1])
    return sum(answer, [])

File: lv1/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
  var answer = [];
  const arr = strings.map((e) => e[n]).sort()
  const arr2 = [...new Set(arr)]
  arr2.forEach((e, i) => {
    const arr3 = []
    strings.forEach((e2, i2) => {
      if (e2[n] === e) {
        arr3.push(e2)
      }
    })
    arr3.sort()
    answer = [...answer, ...arr3.sort()]
  })
  return answer;
}